lib/signs/utilities/messages.ex

```diff
@@ -23,252 +26,291 @@ defmodule Signs.Utilities.Messages do
-              end
-
-            messages ->
-              messages
-          end
-      end
+    cond do
+      match?({:static_text, {_, _}}, sign_config) ->
+        {:static_text, {line1, line2}} = sign_config
+
+        {Content.Message.Custom.new(line1, :top), Content.Message.Custom.new(line2, :bottom)}
+
+      sign_config == :off ->
+        {%Content.Message.Empty{}, %Content.Message.Empty{}}
+
+      match?(%{source_config: {_, _}}, sign) ->
+        Enum.zip([
+          Tuple.to_list(sign.source_config),
+          Tuple.to_list(predictions),
+          Tuple.to_list(alert_status),
+          Tuple.to_list(scheduled),
+          Tuple.to_list(service_status)
+        ])
+        |> Enum.map(fn {config, predictions, alert_status, scheduled, service_status} ->
+          predictions =
+            filter_predictions(predictions, config, sign_config, current_time, scheduled)
+
+          alert_status = filter_alert_status(alert_status, sign_config)
+
+          Signs.Utilities.Predictions.prediction_message(predictions, config, sign) ||
+            service_ended_message(service_status, config) ||
+            alert_message(alert_status, sign, config) ||
+            Signs.Utilities.Headways.headway_message(sign, config, current_time) ||
+            early_am_message(current_time, scheduled, config) ||
+            %Content.Message.Empty{}
+        end)
+        |> List.to_tuple()
+        |> transform_messages()
 
-    messages =
-      if sign_config == :off or match?({:static_text, _}, sign_config),
-        do: messages,
-        else:
-          Signs.Utilities.LastTrip.get_last_trip_messages(
-            messages,
-            service_statuses_per_source,
-            sign.source_config
-          )
+      true ->
+        config = sign.source_config
+
+        predictions =
+          filter_predictions(predictions, config, sign_config, current_time, scheduled)
+
+        alert_status = filter_alert_status(alert_status, sign_config)
 
-    early_am_status =
-      Signs.Utilities.EarlyAmSuppression.get_early_am_state(current_time, scheduled)
+        Signs.Utilities.Predictions.prediction_messages(predictions, config, sign) ||
+          service_ended_messages(service_status, config) ||
+          alert_messages(alert_status, sign, config) ||
+          Signs.Utilities.Headways.headway_messages(sign, config, current_time) ||
+          early_am_messages(current_time, scheduled, config) ||
+          {%Content.Message.Empty{}, %Content.Message.Empty{}}
+    end
+  end
+
+  @spec transform_messages(Signs.Realtime.sign_messages()) :: Signs.Realtime.sign_messages()
+  defp transform_messages(
+         {%Content.Message.Headways.Paging{range: range, route: top_route},
+          %Content.Message.Headways.Paging{range: range, route: bottom_route}}
+       ) do
+    {%Content.Message.Headways.Top{route: combine_routes(top_route, bottom_route)},
+     %Content.Message.Headways.Bottom{range: range}}
+  end
```

Comment:
Do you mean for Ashmont mezzanine, or all mezzanines? The latter would be a pretty substantial change so I'd want to get extra input on that.

Reply:
The thought originated while looking at the Ashmont mezzanine case, but then expanded to all mezzanines. But yeah, it would be a fairly large departure from the status quo. Probably not necessary to worry about doing the Ashmont case I guess - I assume the headway values are typically different considering they're different modes.

As far as the latter goes, it'd be a simple code change anyway so happy to save it for a later discussion

Reply:
I think that's a good follow-up discussion with product, and yeah, it would be an easy change if we decide to do it.